runtime/onert/core/include/exec/train/ITrainableFunction.h

```diff
@@ -32,6 +35,12 @@ class ITrainableFunction
   virtual ~ITrainableFunction() = default;
   virtual void forward(bool training) = 0;
   virtual void backward() = 0;
+
+  // Implement this if LayerScopeTensors is necessary
+  virtual std::optional<backend::train::LayerScopeTensors> registerLayerScopeTensors()
+  {
+    return std::nullopt;
+  }
```

Comment:
`ops/*Layer.cc` will implement this if LayerScope Tensor is necessary.
If this is implemented `LayerScopeTensor` buffer will be managed by `core` TensorManager. 

for example; https://github.com/Samsung/ONE/pull/13486/files#r1753047724